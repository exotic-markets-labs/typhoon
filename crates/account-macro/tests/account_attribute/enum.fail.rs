use {pinocchio::pubkey::Pubkey, typhoon_account_macro::*};

pub const ID: Pubkey = [
    218, 7, 92, 178, 255, 94, 198, 129, 118, 19, 222, 83, 11, 105, 42, 135, 53, 71, 119, 105, 218,
    71, 67, 12, 189, 129, 84, 51, 92, 74, 131, 39,
];

#[account]
pub enum TestState {
    One,
    Two,
}

pub fn main() {}
