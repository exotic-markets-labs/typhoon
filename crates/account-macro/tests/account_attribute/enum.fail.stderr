error: Deriving Pod is not supported for enums
 --> tests/account_attribute/enum.fail.rs:8:1
  |
8 | #[account]
  | ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `bytemuck::Pod` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Invalid account type
 --> tests/account_attribute/enum.fail.rs:8:1
  |
8 | #[account]
  | ^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `account` (in Nightly builds, run with -Z macro-backtrace for more info)
